core/src/main/scala/com/salesforce/op/stages/impl/classification/MultiClassificationModelSelector.scala

```diff
@@ -184,8 +184,13 @@ case object MultiClassificationModelSelector {
   ): ModelSelector[ModelType, EstimatorType] = {
     val modelStrings = modelTypesToUse.map(_.entryName)
     val modelsToUse =
-      if (modelsAndParameters == defaultModelsAndParams || modelTypesToUse != modelNames) modelsAndParameters
+      // if no models are specified use the defaults and filter by the named models to use
+      if (modelsAndParameters.isEmpty) defaultModelsAndParams
         .filter{ case (e, p) => modelStrings.contains(e.getClass.getSimpleName) }
+      // if models to use has been specified and the models have been specified filter the models by the names
+      else if (modelTypesToUse != modelNames) modelsAndParameters
+        .filter{ case (e, p) => modelStrings.contains(e.getClass.getSimpleName) }
+      // else just use the specified models
       else modelsAndParameters
```

Comment:
Is it a total duplicate? Please, please, extract this common functionality to a trait or something.